src/main/java/net/lightbody/bmp/proxy/selenium/CertificateCreator.java

```diff
@@ -268,4 +281,15 @@ public static X509Certificate generateStdSSLServerCertificate(
         return newCert;
     }
 
+    private static String getCn(X500Principal x500Principal) {
+        X500Name x500Name = new X500Name(x500Principal.getName());
+        RDN[] rdns = x500Name.getRDNs(BCStyle.CN);
+        List<String> names = new ArrayList<>();
+        for (RDN rdn : rdns) {
+            String name = IETFUtils.valueToString(rdn.getFirst().getValue());
+            names.add(name);
+        }
+        return names.getFirst();
```

Comment:
This is for Java 21 - not supported yet, shoudl use get(0) - anyway checking the usability of the solution